client/internal/statemachine/readinstance_by_deviceuid.go

```diff
@@ -0,0 +1,47 @@
+package statemachine
+
+import (
+	"context"
+	"fmt"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/url"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/model/statemachine"
+)
+
+type ReadInstanceByDeviceUidInput struct {
+	Uid string // Uid of the device that runs the state machine
+}
+
+func NewReadInstanceByDeviceUidInput(deviceUid string) ReadInstanceByDeviceUidInput {
+	return ReadInstanceByDeviceUidInput{
+		Uid: deviceUid,
+	}
+}
+
+type ReadInstanceByDeviceUidOutput = statemachine.Instance
+
+var NewReadInstanceByDeviceUidOutputBuilder = statemachine.NewInstanceBuilder
```

Comment:
I don't think we need this var. It can potentially cause bugs later as it may be mutated and consumers not know

Reply:
Yea I was thinking about this too, it was a tradeoff between what you have mentioned and consistent interface. I chose the latter because I see more benefits: (1) It follows the same interface convention as other classes, e.g. if I have a method called `readXYZ` then the user can always expect an input and output builder called `readXYZInputBuilder` and `readXYZOutputBuilder`, it makes it easier to use. (2) It hides internal implementation details like state machine instance, the consumer can know what is a state machine because it is widely used computer science term, but in the future we will have `stateMachineDetails`, `stateMachineHooks`, `stateMachineActions`, which is specific to our internal implementation, I do not want to consumer to try and understand that, so I just encapsulate that as generic Input/output. (3) It makes things easier to change, e.g. if we want to change the name of `statemachine.Instance`, we can do that without affecting consumers' code.